storage: retire storage.value_blocks.enabled setting

RaduBerinde opened the pull request:
This setting no longer does anything - for clusters 23.2 and above we
always use the v4 sstable format which always enables value blocks.

Epic: none
Release note (general change): The "storage.value_blocks.enabled"
setting has been retired; value blocks are always enabled.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/122164)
<!-- Reviewable:end -->

jbowens reviewed (approved):
Bit confused about how Essential CI passed and check_generated_code failed, but it looks like we need to regenerate `docs/generated/settings/settings.html`

<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 8 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/122164#-:-NvCuiZzD2sQYpJ4jWZS:b-jghuey)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

RaduBerinde committed:
storage: retire storage.value_blocks.enabled setting

This setting no longer does anything - for clusters 23.2 and above we
always use the v4 sstable format which always enables value blocks.

Epic: none
Release note (general change): The "storage.value_blocks.enabled"
setting has been retired; value blocks are always enabled.

RaduBerinde commented:
TFTR!

RaduBerinde commented:
bors r+
